src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -1,139 +1,2148 @@
-SDCategory: Silverpine Forest
-EndScriptData */
-
-/* ContentData
-npc_deathstalker_erland
-EndContentData */
+/*
+* This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+*
+* This program is free software; you can redistribute it and/or modify it
+* under the terms of the GNU General Public License as published by the
+* Free Software Foundation; either version 2 of the License, or (at your
+* option) any later version.
+*
+* This program is distributed in the hope that it will be useful, but WITHOUT
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+* more details.
+*
+* You should have received a copy of the GNU General Public License along
+* with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
 
+#include "CombatAI.h"
+#include "GameObject.h"
+#include "GameObjectAI.h"
+#include "GridNotifiers.h"
+#include "MotionMaster.h"
+#include "ObjectAccessor.h"
+#include "ObjectMgr.h"
+#include "Object.h"
+#include "Player.h"
 #include "ScriptMgr.h"
 #include "ScriptedCreature.h"
-#include "ScriptedEscortAI.h"
-#include "Player.h"
+#include "ScriptedGossip.h"
+#include "ScriptedCreature.h"
+#include "SpellAuras.h"
+#include "SpellScript.h"
+#include "TemporarySummon.h"
+#include "Unit.h"
+#include "Vehicle.h"
+
+enum QuestTheGilneasLiberationFront
+{
+    NPC_FORSAKEN_TROOPER1 = 44791,
+    NPC_FORSAKEN_TROOPER2 = 44792,
+    NPC_WORGEN_RENEGATE = 44793,
+
+    SPELL_HEARTSTRIKE = 84182,
+    SPELL_DARKENED = 73307,
+    SPELL_FLURRY_OF_CLAWS_CHANNEL = 80365,
+    SPELL_FLURRY_OF_CLAWS = 80367,
+
+    EVENT_CHECK_SHOWFIGHT = 101,
+    EVENT_SPELL_QUEUE = 102,
+    EVENT_SPELL_DAMAGE = 103,
+
+    MOVE_TO_HOMEPOSITION,
+
+    TALK_WORGEN_DEFEATED = 0
+};
+
+// Worgen Renegade - 44793
+struct npc_silverpine_worgen_renegade : public ScriptedAI
+{
+    npc_silverpine_worgen_renegade(Creature* creature) : ScriptedAI(creature) { }
+
+    void Reset() override
+    {
+        _events.Reset();
+
+        me->RemoveAura(SPELL_DARKENED);
+
+        _events.ScheduleEvent(EVENT_CHECK_SHOWFIGHT, 1s);
+    }
+
+    void SpellHit(Unit* caster, SpellInfo const* spell) override
+    {
+        if (!caster)
+            return;
+
+        if (spell->Id == SPELL_HEARTSTRIKE)
+            me->AddAura(SPELL_DARKENED, me);
+    }
+
+    void MovementInform(uint32 type, uint32 pointId) override
+    {
+        if (type == POINT_MOTION_TYPE && pointId == MOVE_TO_HOMEPOSITION)
+            _events.ScheduleEvent(EVENT_CHECK_SHOWFIGHT, 25ms);
+    }
+
+    void JustEngagedWith(Unit* /*who*/) override
+    {
+        _events.ScheduleEvent(EVENT_SPELL_QUEUE, 3s);
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _events.Update(diff);
+
+        while (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_CHECK_SHOWFIGHT:
```

Comment:
Try with deactivating gridunload.

```
#
#    GridUnload
#        Description: Unload grids to save memory. Can be disabled if enough memory is available
#                     to speed up moving players to new grids.
#        Default:     1 - (enable, Unload grids)
#                     0 - (disable, Do not unload grids)

GridUnload = 0
```

Reply:
Tested, the creatures do not find each other even though their factions are hostile to each other.

Reply:
Fixed.